src/Sfa.Tl.Matching.Web/Controllers/OpportunityController.cs

```diff
@@ -152,6 +154,8 @@ public async Task<IActionResult> EmailSentReferrals(int id)
         {
             var opportunity = await _opportunityService.GetOpportunity(id);
 
+            await _referralService.SendProviderReferralEmail(id);
```

Comment:
This need to happen before we redirect to Email Sent Screen i.e. in the post of CheckAnswers
Also you might want to move this call to Opportunity service